src/DenielWorld/VanillaCommands/command/PlaySound.php

```diff
@@ -18,123 +19,75 @@ class PlaySound extends PluginCommand implements PluginIdentifiableCommand{
-                if($this->sound_manager->isValidName($args[0])){
-                    $packet = new PlaySoundPacket();
-                    $packet->soundName = $args[0];
-                    $packet->x = $args[2];
-                    $packet->y = $args[3];
-                    $packet->z = $args[4];
-                    $packet->volume = $args[5];
-                    $packet->pitch = $args[6];
-                    $sender->sendDataPacket($packet);
-                }
-                elseif(!$this->sound_manager->isValidName($args[0]) and $this->sound_manager->isValidConstString($args[0])){
-                    $packet = new PlaySoundPacket();
-                    $packet->soundName = $this->sound_manager->constAsName($args[0]);
-                    $packet->x = $args[2];
-                    $packet->y = $args[3];
-                    $packet->z = $args[4];
-                    $packet->volume = $args[5];
-                    $packet->pitch = $args[6];
-                    $sender->sendDataPacket($packet);
-                }
-                else {
-                    $sender->sendMessage(TF::RED . "Invalid sound");
-                }
+    public function execute(CommandSender $sender, string $commandLabel, array $args) {
+        if(count($args) < 1) {
+            throw new InvalidCommandSyntaxException();
+        }
+        if(is_int($args[0]))
+            $name = $this->sound_manager->getNameFromId($args[0]);
+        else
+            $name = $args[0];
+        if(!isset($args[1])){
+            if(!$sender instanceof Player) {
+                $sender->sendMessage("Please run this command in-game");
+                return;
             }
-            else {
-                $sender->sendMessage(TF::RED . "Please provide valid arguments");
+            $packet = new PlaySoundPacket();
+            $packet->soundName = $name;
+            $packet->x = $sender->x;
+            $packet->y = $sender->y;
+            $packet->z = $sender->z;
+            $packet->volume = 3;
+            $packet->pitch = 2;
+        }elseif(!isset($args[2])) {
+            $player = $this->getPlugin()->getServer()->getPlayer($args[1]);
+            if(!$player instanceof Player) {
+                $sender->sendMessage(TextFormat::RED."Player not found");
+                return;
             }
+            $packet = new PlaySoundPacket();
+            $packet->soundName = $name;
+            $packet->x = $player->x;
+            $packet->y = $player->y;
+            $packet->z = $player->z;
+            $packet->volume = 3;
+            $packet->pitch = 2;
+            $player->sendDataPacket($packet);
+        }elseif(count($args) == 5) {
+            $packet = new PlaySoundPacket();
+            $packet->soundName = $name;
+            $packet->x = $args[2];
```

Comment:
Instead of repeating all this over and over again, i suggest something like (warning, sloppy code)
```
( isset(args[2]) && isset(args[3]) && isset(args[4]) ) ? $v3 = new Vector3(args2,args3,args4) : $v3 = $player->asVector3();
```